### DISCUSSION TITLE
tasmota factory reset not working

### DISCUSSION THREAD

**Discussion by phantu91git** [arendst/Tasmota#21912] [Support request] [answered]
hi everyone, i have building tasmota custom firmware.
In the user_config_override.h:
#define SAVE_DATA              600                 // [SaveData] Save changed parameters to Flash (0 = disable, 1 - 3600 seconds)                                                                                                                 #define SAVE_STATE             true              // [SetOption0] Save changed power state to Flash (false = disable, true = enable)                                                                                                               #define BOOT_LOOP_OFFSET       7                 // [SetOption36] Number of boot loops before starting restoring defaults (0 = disable, 1..200 = boot loops offset)
i have defined the boot_loop_offset but after i flashed my esp. i tried to default factory reset with this procedures:
1. power devices for more than 40s
2. repeatedly turn on device for 7 times and at the 8 times i let it on.(try other 7 time too)
but the device does not factory reset. So is it need diffirent configuration or i made an mistake somewhere.
thank you all

**Comment by sfromis** [comment]
Typical issue when failing to get the quick power cycle recovery working is being too quick. Needs depend on your hardware, but as a starting point, I recommend 7 cycles of 2 seconds off, and two seconds on before next cycle. And make sure that you do not have `SetOption65` ON to disable.

**Reply by phantu91git** [reply]
i tried quite time with your suggestion. 3 times (7 cycles of 2 second off, 2 sec on with my mobile timer). my configuration does not have setoption65 option, i think it mean default value. it work with orginal firmware and my esp is esp12s.
have no ideal why it not default reset, gonna to re-compile with default option for boot_loop and report back to figure out the solution

**Reply by phantu91git** [reply, marked as answer]
i tested quite few time with the orginal firmware and custom build.
with tx,rx tie to some sensor, mostly time it not work. and maybe someone have better luck with sfromis suggesstion.
the better solution should be using "button 1" as documentation.

**Comment by tilator** [comment]
There is still something odd about resetting it. Powercycling it 7 times quicly brings up ESP-04D09D AP, but it disappears before it can be connected. Most networks are not fast enough to see it at all before it's gone.

Is there some setting not reset properly that might make the AP to disappear some seconds after it it set up?

**Reply by sfromis** [reply]
ESP-xxxxxx would be useless to connect to if you could, as it is not from Tasmota. What happens is rather unclear.

BTW. "Quickly" power cycling may be too fast, usually I recommend each half-step be two seconds.

**Reply by tilator** [reply]
You are quite probably write. I have used this device so long ago, I'm not sure what FW it has installed.
